corehq/motech/repeaters/views/repeaters.py

```diff
@@ -44,25 +45,36 @@ class DomainForwardingOptionsView(BaseAdminProjectSettingsView):
     def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
-    @property
-    def repeater_types_info(self):
+    def get_repeater_types_info(self, state_counts):
+        def get_repeaters_with_state_counts(repeater_class):
+            repeaters = repeater_class.objects.by_domain(self.domain)
+            for repeater in repeaters:
+                assert not hasattr(repeater, "count_State"), repeater.count_State
+                repeater.count_State = {s.name: state_counts[repeater.id][s] for s in State}
+            return repeaters
+
         return [
             RepeaterTypeInfo(
                 class_name=r._repeater_type,
                 friendly_name=r.friendly_name,
                 has_config=r._has_config,
-                instances=r.objects.by_domain(self.domain),
+                instances=get_repeaters_with_state_counts(r),
             )
             for r in get_all_repeater_types().values()
             if r.available_for_domain(self.domain)
         ]
 
     @property
     def page_context(self):
+        state_counts = SQLRepeatRecord.objects.count_by_repeater_and_state(domain=self.domain)
         return {
             'report': 'repeat_record_report',
-            'repeater_types_info': self.repeater_types_info,
-            'pending_record_count': SQLRepeatRecord.objects.count_pending_records_for_domain(self.domain),
+            'repeater_types_info': self.get_repeater_types_info(state_counts),
+            'pending_record_count': sum(
+                count for repeater_id, states in state_counts.items()
+                for state, count in states.items()
+                if state == State.Pending or state == State.Fail
```

Comment:
Yes, _failed_ records will be retried, so they are also pending. In case you're curious, `Cancelled` is the state a record will be in if it failed too many times and will no longer be retried.

Reply:
So is that a change to the functionality? Users were seeing just count on "pending" records but now they would see "pending" + "fail"?

Reply:
No, it is not a change in functionality. Previously this used `count_pending_records_for_domain`:

https://github.com/dimagi/commcare-hq/blob/e1286bf543a96aad43887e46e97b552a788d87d8/corehq/motech/repeaters/models.py#L1353-L1354

which filtered on `next_check__isnull=False`. The check constraint on `next_check` enforces a non-null value for records with states `Pending` and `Fail`.

https://github.com/dimagi/commcare-hq/blob/e1286bf543a96aad43887e46e97b552a788d87d8/corehq/motech/repeaters/models.py#L1442-L1446

Reply:
Thank you @millerdev 

> which filtered on next_check__isnull=False. The check constraint on next_check enforces a non-null value for records with states Pending and Fail.

That connection was not obvious to me from my read.